README.md

```diff
@@ -20,212 +20,324 @@ To install shelf-lib-js for development, run the following commands:
+* `strictHostCheck` `{boolean}` - Turns off request strictHostCheck. This defaults to `true`.
+* `timeoutDuration` `{int}` - Sets the amount of time in milliseconds before requests timeout. This defaults to 300000 (five minutes).
+* `retries` `{int}` - The number of times we should retry the request. This functionality applies to TCP level errors (socket timeout for example) and anything within the 500 HTTP status code range.
 
 Example:
 
     var libOptions = {
-        logLevel: "debug",          // Sets the desired amount of logging.
-                                    // Values can be: "info", "debug", or "warning".
-                                    // Defaults to "warning".
-        "strictHostCheck": false    // Turns off request strictHostCheck.
-                                    // Defaults to true.
+        logLevel: "debug",
+        strictHostCheck: false
     };
 
-    var shelfLib = require("shelf-lib")("<URL where shelf is hosted>", libOptions);
+    var shelfLib = require("shelf-lib")("https://api.shelf.com", libOptions);
 
 
-Reference Creation
-------------------
+### `shelfLib~initReference(refName, authToken)`
 
-    // Grab a reference to a bucket.
-    var reference = shelfLib.initReference("refName", "<super secret pyshelf API key>");
+* `refName` `{string}` - Could also be called "bucket", "storage" or "shelf". This represents a specific storage space.
+* `authToken` `{string}` - The authentication token used to authenticate with the `reference`
+* Returns: `{Object}` - A [reference](#reference) object.
 
+Example:
 
-Artifact Reference Creation
----------------------------
-
-    // Grab the artifact reference at location "/path".
-    var artifact = reference.initArtifact("path");
+    var reference = shelfLib.initReference("shadow", "bDMKbnnzrDlmNs34NHGkvHOWDVdE4okV");
 
 
-Uploading
+reference
 ---------
 
-To upload an artifact with the contents from a variable:
+Represents what people might call a "reference name", "bucket", "storage" or "shelf". See [the Shelf documentation](https://github.com/not-nexus/shelf/blob/master/docs/configuration.md) for more information.
 
-    artifact.upload("Hello world!").then((uploadLocation) => {
-        console.log("Uploaded content to: " + uploadLocation);
-    }, (err) => {
-        console.log("Hit an error: " + err);
-    });
 
-To upload an artifact from a file:
+### `reference~initArtifact(path)`
 
-    artifact.uploadFromFile("./file-to-upload.txt").then((uploadLocation) => {
-        console.log("Uploaded content to: " + uploadLocation);
-    }, (err) => {
-        console.log("Hit an error: " + err);
-    });
+* `path` `{string}` - Just the path to the artifact inside of the reference. It is `<path>` in `https://api.shelf.com/shadow/artifact/<path>`.
```

Comment:
This one is much more important because it gives context to people who know how to use shelf.